### ISSUE TITLE
Current status and future plans

### ISSUE THREAD

**Issue by LecrisUT** [pboettch/json-schema-validator#261]
Is there an overview of the current status, e.g. what schema versions are supported, and if there are any future plans. Is the issue that `nlohmann::json` is a header-only library why this one was not incorporated directly there? Maybe it is worth testing the waters upstream since cmake has come a long way and header-only formats are starting to become unfashionable.

**Comment by pboettch** [comment]
Draft7 is supported. Due to a lack of time and motivation, there currently no plan that I'd work on other versions.

**Comment by LecrisUT** [comment]
Would you be able to make a break-down of the issues needed for the other drafts, mostly the big redesign roadblocks needed, e.g. the relative path parser.

**Comment by pboettch** [comment]
No, I don't know about the new versions of the standard and their specifics. 

Relative path parsers is already an issue with this library.

**Comment by LecrisUT** [comment]
If we use the latest version of the test-suite, would it reflect all of the new specifications, relative path, etc.?

**Comment by LecrisUT** [comment]
Ok, maybe not the drafts version, but the test-suite repo version. I.e. within the draft, is it complete to cover all use-cases or are there some things that need to be implementation dependent

**Comment by LecrisUT** [comment]
Indeed I understand that, I was asking of e.g. [draft7](https://github.com/json-schema-org/JSON-Schema-Test-Suite/tree/main/tests/draft7) tests if itself is complete. I see that they are complete with respect to previous ones, e.g. both draft4 and draft7 have tests for `AnyOf`. But is each one complete in that they cover any logic, constraints, navigation, anything that you would encounter when designing and using a json schema? E.g. for the [`#ref`](https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/main/tests/draft7/ref.json) syntax does it cover different types of path sources and all of the various places where `#ref` is legal to be used?

That is because there are individual issue test checks, but shouldn't those be accounted in the `main` branch of the test-suite?

**Comment by pboettch** [comment]
> Indeed I understand that, I was asking of e.g. [draft7](https://github.com/json-schema-org/JSON-Schema-Test-Suite/tree/main/tests/draft7) tests if itself is complete. I see that they are complete with respect to previous ones, e.g. both draft4 and draft7 have tests for `AnyOf`. But is each one complete in that they cover any logic, constraints, navigation, anything that you would encounter when designing and using a json schema? E.g. for the [`#ref`](https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/main/tests/draft7/ref.json) syntax does it cover different types of path sources and all of the various places where `#ref` is legal to be used?

No, not everything is/was covered. Hence the issues I had over the years and fixes I had to do. A lot of $ref-stuff was not working correctly in early version of this library because of the test-suite being too nice. Now it's better, but I'm not sure it covers everything.
 
> That is because there are individual issue test checks, but shouldn't those be accounted in the `main` branch of the test-suite?

Maybe. But some are specific to the C++ basic type limitations/ranges (uint64_t, double).

**Comment by pboettch** [comment]
Before doing any new work on the validator concerning newer drafts, maybe we should wait:  https://json-schema.org/blog/posts/future-of-json-schema

**Comment by LecrisUT** [comment]
Wouldn't those still be built on top of the current drafts?

**Comment by pboettch** [comment]
Surely, but there will changes. No need to code stuff that'll be removed.
